### PR TITLE
Show DEBUG in knowhere version for debug build

### PR DESCRIPTION
**Pull request by cydrain**
Issue: #844

### CODE REVIEW

**Comment by cydrain** [comment]
test with Milvus debug build, can see following message:
```
I20230506 11:35:00.246528 51480 knowhere_config.cc:47] [KNOWHERE][ShowVersion][milvus] Knowhere Version: 6448cf09 (DEBUG)
```

File: src/common/comp/knowhere_config.cc [hunk 1]
```diff
@@ -24,21 +24,27 @@
...
-#else
-    LOG_KNOWHERE_INFO_ << "Knowhere Version: " << XSTR(KNOWHERE_VERSION);
+    msg = msg + "-gpu";
 #endif
 #else
-    LOG_KNOWHERE_INFO_ << "Knowhere Version: unknown";
+    msg = msg + " unknown";
+#endif
+
+#ifdef _DEBUG
```
Comment (PwzXxm) [inline src/common/comp/knowhere_config.cc:44]: AFAIK, `_DEBUG` is not in the standard, but `NDEBUG` is. Can you double-check?

Comment (cydrain) [reply src/common/comp/knowhere_config.cc]: ok, updated

**Pull request merged**